### DISCUSSION TITLE
Documentation for installing OLS on an AstroArch

### DISCUSSION THREAD

**Discussion by Pti-Jean** [artyom-beilis/OpenLiveStacker#92] [Q&A]
(Message translated into English)
Hello,
Can you create a documentation to install OpenLiveStacker on an AstroArch:
https://github.com/devDucks/astroarch
Because this documentation is missing!

(Original message in French)
Bonjour,
Pouvez-vous créer une documentation pour installer OpenLiveStacker sur une AstroArch:
https://github.com/devDucks/astroarch
Car cette documentation manque!
JM

**Comment by artyom-beilis** [comment]
Generally there are build instructions here: https://github.com/artyom-beilis/OpenLiveStacker?tab=readme-ov-file#using-on-linux

I'm not familiar with arch but other than probably little bit different package names it should be the same.

**Comment by Pti-Jean** [comment]
(Message translated into English):
So yes, I got OLS working on AstroArch...
Instead of the ‘apt-get install’, I replaced it with:
`$ sudo pacman -S cmake libcurl-compat libcurl-gnutls opencv python-opencv base-devel`
then, I followed the doc... I don't understand everything about OLS, but it seems to work... and even on an RPI5...
All that's missing is .fits support for it to be complete!

(Original message in French):
Donc oui, j'ai fait fonctionner OLS sur AstroArch...
À la place de l'"apt-get install", j'ai remplacé par:
`$ sudo pacman -S cmake libcurl-compat libcurl-gnutls opencv python-opencv base-devel`
puis, après, j'ai suivi la doc... Je ne comprends pas tout à OLS, mais il semble fonctionner... et même sur un RPI5...
Manque plus que le support des .fits, pour qu'il soit complet!

JM

**Comment by artyom-beilis** [comment]
> Instead of the ‘apt-get install’, I replaced it with:
> 
>     $ sudo pacman -S cmake libcurl-compat libcurl-gnutls opencv python-opencv base-devel

Excellent. Can you check other dependencies so I can extend the docs: libraw, libuvc/libusb, libgphoto2 

> All that's missing is .fits support for it to be complete!

What camera do you have? Maybe it is supported with internal driver, can be easily added?

Can you request saving files as tiff? Since tiff has same capabilities as fits but much more universal

**Comment by Pti-Jean** [comment]
@artyom-beilis Excellent. Can you check other dependencies so I can extend the docs: libraw, libuvc/libusb, libgphoto2
(en) How do you check?
(fr) Comment vérifier ?

@artyom-beilis What camera do you have? Maybe it is supported with internal driver, can be easily added?
(en) It's a ZWO ASI533MC colour camera...
(en) But I posted at the end of this thread:
https://github.com/artyom-beilis/OpenLiveStacker/issues/74
(fr) C'est-une caméra couleur ZWO ASI533MC...
(fr) Mais j'ai posté à la fin de ce fils de discussion:
https://github.com/artyom-beilis/OpenLiveStacker/issues/74

@artyom-beilis Can you request saving files as tiff? Since tiff has same capabilities as fits but much more universal
(en) Except that Kstars, which is the reference in astro in the Linux world, does not allow saving in tiff format! The default format is fits!
(fr) Sauf que Kstars qui est la référence en astro dans le monde Linux, ne permet pas d’enregistrer au format tiff! Le format par défaut est le fits!

 JM

**Comment by artyom-beilis** [comment]
> (en) It's a ZWO ASI533MC colour camera...

Just build OpenLiveStacker with ASI ZWO SDK, it is naively supported camera. It would work much better with direct connection - you'll have control over camera settings and can manage profiles inside OpenLiveStacker

**Comment by Pti-Jean** [comment]
> Just build OpenLiveStacker with ASI ZWO SDK, it is naively supported camera. It would work much better with direct connection - you'll have control over camera settings and can manage profiles inside OpenLiveStacker

(en):
Except that OpenLiveStacker in this case will come into conflict with KStars, Ekos... and I'd absolutely like to avoid that... What's more, I want to keep the files before stacking with the crucial information given by KStars and Ekos in the file names!
So, the solution of managing the camera live... I think that's a poor one!

(fr):
Sauf que OpenLiveStacker dans ce cas va rentrer en conflit avec KStars, Ekos... et cela j'aimerais absolument l'éviter... De plus, je tiens à garder les fichiers avant stacking avec les infos cruciales données par KStars, Ekos dans le nom des fichiers!
Donc, la solution de gérer la caméra en direct... je trouve cela moyen!

JM

**Comment by artyom-beilis** [comment]
Once again. I understand your concerns but currently using OLS directly with camera is way better solution (at least from OLS point of view) 

OLS also can save files and meta data about camera setup (it saves it together with result file)

**Comment by Pti-Jean** [comment]
> OLS also can save files and meta data about camera setup (it saves it together with result file)

(en):
The information communicated in the file name is information in addition to that in the metadata, such as the gain of the camera, the name of the star being photographed... and even the exposure time, it's easier to read it in the file name than to look for it in the metadata!

(fr):
Les infos communiquées dans le nom du fichier sont des infos en plus que celle des métadonnées, comme le gain de la caméra, le nom de l'astre photographier... et même le temps d'exposition, c'est plus simple de le lire dans le nom du fichier, que d'aller le cherché dans les métadonnées!

JM

**Comment by artyom-beilis** [comment]
Here an example of txt file saved along with the jpeg and tiff file

```
$ cat M20_20230620_231116_stacked_v5.txt
Name           M20_20230620_231116
Time           2023-06-20 23:15:45
Frames         52
Integration (s)260
Frame:
 - format raw16
 - width  1304
 - height 976
 - bin    1
 - mono   0
Stackig:
 - remove_satellites  0
 - derotate           1
 - derotate_mirror    0
Target:
 - RA 270.675
 - DE -22.9828
Geolocation
 - lat 31.88
 - lon 34.8
Camera Settings:
 - Auto Exp.: 0
 - Auto WB: 0
 - Exp.: 5000
 - WB Red: 52
 - WB Blue: 95
 - Gain: 300
 - Temp.: 29.1
```

I hope it answers your current needs.

Anyway firs support is something that can be added

**Comment by Pti-Jean** [comment]
(en):
Even if it's not as convenient as the file name... you still need to be able to make the link between the .txt file and the source image files... because if you've photographed several stars, you need to be able to find your way around!

(fr):
Même si c'est pas aussi commode que dans le nom du fichier... faudrait quand-même pouvoir faire le lien entre le fichier .txt et les fichiers images sources... car dans le cas où tu as photographié plusieurs astres, il faut pouvoir s'y retrouver!

JM

**Comment by artyom-beilis** [comment]
> Even if it's not as convenient as the file name... you still need to be able to make the link between the .txt file

They have _exactly_ the same name just one ends with .jpeg and other .txt

**Comment by Pti-Jean** [comment]
(en):
The source image file names are in .jpg ... We must be losing a lot of quality if we don't want to do some serious processing with Siril afterwards?
Yes, because I like to see the result live in Live Stacking, so that when I get home I can do some serious processing with serious software like Siril, so that I can share it on the Internet!

(fr):
Les noms des fichiers images sources sont en .jpg ... On doit perdre grave en qualité, si pas la suite on souhaite faire un traitement sérieux avec Siril ??
Oui, parce que moi, j'aime bien visualiser le résultat en direct en Live Stacking, pour une fois rentré chez moi, faire un traitement sérieux avec un logiciel sérieux comme Siril, pour pouvoir le partager sur Internet!

JM

**Comment by artyom-beilis** [comment]
OpenLiveStacker saves:

- jpeg - exact result of what you have seen in live - it is ready for sharing
- tiff - 16 bit stacked file not stretched, no white balance - so you can do any post processing you wanted in GIMP or other tools
- txt - description of what have done

If you select "save all files" checkbox during stacking it would save _original_ files as well (as tiff) so you can stack them on your own.

**Comment by Pti-Jean** [comment]
(en):
So there would be a way to make the link between the .txt file and the corresponding .tiff files?

(fr):
Et donc, il y aurait un moyen de faire le lien entre le fichier .txt et les fichiers .tiff correspondent ?

JM

**Comment by artyom-beilis** [comment]
> So there would be a way to make the link between the .txt file and the corresponding .tiff files?

Obviously

**Comment by artyom-beilis** [comment]
fits support implemented, run git pull and update libcfitsio is new dependency for fits support
